manager/manager/manager.go

```diff
@@ -113,15 +113,16 @@ func (m *manager) CreateDeployment(t *Template) (*Deployment, error) {
 		return nil, err
 	}
 
-	// TODO: Mark this as failed instead of deleting.
 	if err := m.deployer.CreateConfiguration(et.Config); err != nil {
-		m.repository.DeleteDeployment(t.Name, true)
+		// Deployment failed, mark as deleted
+		log.Printf("CreateConfiguration failed: %v", err)
+		m.repository.SetDeploymentStatus(t.Name, FailedStatus)
```

Comment:
We will probably want a list of errors at the deployment level as well when you follow up to add error messages.